lib/OpModel/TTNN/Conversion.cpp

```diff
@@ -0,0 +1,160 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "TTNNOpModel.h"
+
+#ifdef TTMLIR_ENABLE_OPMODEL
+#include "Conversion.hpp"
+#include "ttmlir/Dialect/TT/Utils/CoreRangeSet.h"
+
+namespace mlir::tt::op_model::ttnn {
+
+namespace conversion {
+
+::tt::tt_metal::DataType
+getDataType(const mlir::tt::ttnn::TTNNLayoutAttr layout) {
+  auto dataType = layout.getDataType();
+
+  switch (dataType) {
+  case tt::DataType::Float32:
+    return ::tt::tt_metal::DataType::FLOAT32;
+  case tt::DataType::BFloat16:
+    return ::tt::tt_metal::DataType::BFLOAT16;
+  case tt::DataType::BFP_BFloat8:
+    return ::tt::tt_metal::DataType::BFLOAT8_B;
+  case tt::DataType::BFP_BFloat4:
+    return ::tt::tt_metal::DataType::BFLOAT4_B;
+  case tt::DataType::UInt32:
+    return ::tt::tt_metal::DataType::UINT32;
+  case tt::DataType::UInt16:
+    return ::tt::tt_metal::DataType::UINT16;
+  case tt::DataType::UInt8:
+    return ::tt::tt_metal::DataType::UINT8;
+  default:
+    throw std::runtime_error("Invalid element type");
+  }
+}
+
+::ttnn::SimpleShape getSimpleShape(const ::llvm::ArrayRef<int64_t> &shape) {
+  ::tt::tt_metal::SmallVector<uint32_t> small_vector_shape;
+  for (const auto &dim : shape) {
+    small_vector_shape.push_back(static_cast<uint32_t>(dim));
+  }
+
+  return ::ttnn::SimpleShape(small_vector_shape);
+}
+
+const std::array<uint32_t, 2>
+getShardShape(const mlir::tt::ttnn::TTNNLayoutAttr &layout) {
+  const auto layoutShardTile = layout.getScalarShardShape();
+
+  if (layoutShardTile.size() != 2) {
+    llvm::errs() << "ERROR: layout_shard_tile.size() != 2\n";
+    return {0, 0};
+  }
+
+  std::array<uint32_t, 2> shardShape;
+  shardShape[0] = layoutShardTile[0];
+  shardShape[1] = layoutShardTile[1];
+  return shardShape;
+}
+
+::tt::tt_metal::Layout
+getPageLayout(const mlir::tt::ttnn::TTNNLayoutAttr &layout) {
+  return layout.isTiled() ? ::tt::tt_metal::Layout::TILE
+                          : ::tt::tt_metal::Layout::ROW_MAJOR;
+}
+
+::tt::tt_metal::CoreRangeSet
+getCoreRangeSet(const mlir::tt::ttnn::TTNNLayoutAttr &layout,
+                const mlir::tt::GridAttr &workerGrid) {
+  std::set<::tt::tt_metal::CoreRange> coreRangeSet;
+  for (const auto &[loc, size] :
+       utils::toCoreRangeSet(layout.getGrid(), workerGrid)) {
+    coreRangeSet.insert(::tt::tt_metal::CoreRange(
+        CoreCoord(loc[0], loc[1]),
+        CoreCoord(loc[0] + size[0] - 1, loc[1] + size[1] - 1)));
+  }
+  return ::tt::tt_metal::CoreRangeSet(coreRangeSet);
+}
+
+std::optional<::tt::tt_metal::ShardSpec>
+getShardSpec(const mlir::tt::ttnn::TTNNLayoutAttr &layout,
+             const mlir::tt::GridAttr &workerGrid) {
+  // tt_ShardOrientation is not part of ttnn::TTNNLayoutAttr;
+  // defaulting to ROW_MAJOR. TODO: figure out if we need to expose this
```

Comment:
`ShardSpec` defines the orientation. There's an outstanding ticket with the goal of adding a `ShardSpecAttr` to `MemoryConfigAttr`.

https://github.com/tenstorrent/tt-mlir/issues/620

Reply:
thanks, added todo